apps/SageAccountWeb/src/components/TwoFactorAuth/ResetTwoFactorAuth.tsx

```diff
@@ -98,16 +117,25 @@ export function ResetTwoFactorAuth() {
                     You can re-activate two-factor authentication at any time in
                     your account settings.
                   </Typography>
+                  {showPasswordField && (
+                    <PasswordField
+                      value={password}
+                      onChange={e => setPassword(e.target.value)}
+                    />
+                  )}
                   <Button
                     fullWidth
                     variant={'contained'}
                     onClick={() => {
                       disableTwoFactorAuth({
                         ...twoFaRequiredToken,
                         twoFaResetToken: token,
+                        password: password,
                       })
                     }}
-                    disabled={isSuccess || isPending}
+                    disabled={
+                      isSuccess || isPending || (showPasswordField && !password)
+                    }
```

Comment:
Disable the button if the user requested to disable 2FA with a password and has not entered a password